From 0d4950c6ca519f12d392a087d3aa4bf55d91903e Mon Sep 17 00:00:00 2001
From: Yinhang Liu <yinhang.liu@intel.com>
Date: Thu, 11 May 2023 12:30:11 +0800
Subject: [PATCH 115/205] msdk: Fix initialization of the msdk_session.impl_idx

This fixes the issue of msdk_session.impl_idx not being initialized.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4448>
---
 sys/msdk/msdk.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sys/msdk/msdk.c b/sys/msdk/msdk.c
index c8237d1aa..ad4a70d37 100644
--- a/sys/msdk/msdk.c
+++ b/sys/msdk/msdk.c
@@ -354,6 +354,7 @@ msdk_init_msdk_session (mfxIMPL impl, mfxVersion * pver,
 
   msdk_session->session = session;
   msdk_session->loader = NULL;
+  msdk_session->impl_idx = 0;
 
   return MFX_ERR_NONE;
 }
@@ -403,6 +404,7 @@ msdk_open_session (mfxIMPL impl)
 
   msdk_session.session = NULL;
   msdk_session.loader = NULL;
+  msdk_session.impl_idx = 0;
   status = msdk_init_msdk_session (impl, &version, &msdk_session);
 
   if (status != MFX_ERR_NONE)
@@ -433,6 +435,7 @@ failed:
   msdk_close_session (&msdk_session);
   msdk_session.session = NULL;
   msdk_session.loader = NULL;
+  msdk_session.impl_idx = 0;
   return msdk_session;
 }
 
-- 
2.40.1

